amy/emails/tests/actions/test_instructor_training_approaching_update_receiver.py

```diff
@@ -0,0 +1,398 @@
+    instructor_training_approaching_signal,
+    instructor_training_approaching_update_signal,
+)
+from workshops.models import Event, Organization, Person, Role, Tag, Task
+from workshops.tests.base import TestBase
+
+
+class TestInstructorTrainingApproachingUpdateReceiver(TestCase):
+    def setUp(self) -> None:
+        self.ttt_organization = Organization.objects.create(
+            domain="carpentries.org", fullname="Instructor Training"
+        )
+        self.event = Event.objects.create(
+            slug="test-event",
+            host=Organization.objects.create(domain="example.com", fullname="Example"),
+            administrator=self.ttt_organization,
+            start=date.today() + timedelta(days=30),
+        )
+        ttt_tag = Tag.objects.create(name="TTT")
+        self.event.tags.add(ttt_tag)
+        instructor_role = Role.objects.create(name="instructor")
+        self.instructor1 = Person.objects.create(
+            personal="Test", family="Test", email="test1@example.org", username="test1"
+        )
+        self.instructor2 = Person.objects.create(
+            personal="Test", family="Test", email="test2@example.org", username="test2"
+        )
+        Task.objects.create(
+            event=self.event, person=self.instructor1, role=instructor_role
+        )
+        Task.objects.create(
+            event=self.event, person=self.instructor2, role=instructor_role
+        )
+
+    @patch("workshops.utils.feature_flags.logger")
+    def test_disabled_when_no_feature_flag(self, mock_logger) -> None:
+        # Arrange
+        request = RequestFactory().get("/")
+        with self.settings(FLAGS={"EMAIL_MODULE": [("boolean", False)]}):
+            # Act
+            instructor_training_approaching_update_receiver(None, request=request)
+            # Assert
+            mock_logger.debug.assert_called_once_with(
+                "EMAIL_MODULE feature flag not set, skipping "
+                "instructor_training_approaching_update_receiver"
+            )
+
+    def test_receiver_connected_to_signal(self) -> None:
+        # Arrange
+        original_receivers = instructor_training_approaching_update_signal.receivers[:]
+
+        # Act
+        # attempt to connect the receiver
+        instructor_training_approaching_update_signal.connect(
+            instructor_training_approaching_update_receiver
+        )
+        new_receivers = instructor_training_approaching_update_signal.receivers[:]
+
+        # Assert
+        # the same receiver list means this receiver has already been connected
+        self.assertEqual(original_receivers, new_receivers)
+
+    @override_settings(FLAGS={"EMAIL_MODULE": [("boolean", True)]})
+    def test_action_triggered(self) -> None:
+        # Arrange
+        request = RequestFactory().get("/")
+
+        template = EmailTemplate.objects.create(
+            name="Test Email Template",
+            signal=instructor_training_approaching_update_signal.signal_name,
```

Comment:
Another case where I was expecting this to be `instructor_training_approaching_signal.signal_name`

Reply:
Changed to a constant for the signal name.